flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnConfigurationITCase.java

```diff
@@ -208,4 +215,12 @@ private boolean hasTaskManagerConnectedAndReportedSlots(Collection<TaskManagerIn
 			return taskManagerInfo.getNumberSlots() > 0;
 		}
 	}
+
+	private static int calculateManagedMemorySizeMB(Configuration originalConfiguration, int numSlotsPerTaskManager) {
+		Configuration configuration = new Configuration(originalConfiguration); // copy, because we alter the config
```

Comment:
It is a bit weird that we have to do something like this in a test. This might indicate a design flaw of the `YarnResourceManager#updateTaskManagerConfigAndCreateWorkerSlotProfiles` method.